Sources/SwiftDocC/Utility/Weak.swift

```diff
@@ -0,0 +1,14 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+/// A wrapper that provides weak ownership of a value.
+struct Weak<T: AnyObject> {
```

Comment:
Is there any mechanism that drains the shared pool or otherwise removes unused elements? If not, I think we'll still end up with a shared pool full of empty weak boxes.

Reply:
I was worried about that too, but the added logic in the `deinit` of the `Storage` should handle this.

The test that asserts on the emptiness of the cache should verify this.